src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.SearchScope.cs

```diff
@@ -112,11 +112,24 @@ public override SymbolReference CreateReference<T>(SearchResult<T> searchResult)
                     _metadataReference);
             }
 
-            protected override Task<IEnumerable<ISymbol>> FindDeclarationsAsync(string name, SymbolFilter filter, SearchQuery searchQuery)
+            protected override async Task<IEnumerable<ISymbol>> FindDeclarationsAsync(string name, SymbolFilter filter, SearchQuery searchQuery)
             {
-                return SymbolFinder.FindDeclarationsAsync(
+                // when searching metadata, give ourselves a limit on how long the search can take.
+                // That way AddImport doesn't pause for a long time while producing metadata indices.
+                const int SearchTimeoutMS = 50;
```

Comment:
That seems surprisingly short -- you sure you want that? Also, if this fails, can we collect some telemetry or equivalent so we can understand how often we have failed the user at being awesome?